### DISCUSSION TITLE
Visualizer feature: access to per-broker tariffs and subscriptions

### DISCUSSION THREAD

**Discussion by jecollins** [powertac/powertac-server#1174] [General]
The current viz shows the raw number of subscriptions and net numbers of subscription signups/withdraws for each broker over time. This is a very coarse representation of what's going on in the tariff market. We cannot tell whether the subscriptions are households, EVs, solar producers, or commercial/industrial customers. We also cannot tell which brokers are offering storage tariffs or production tariffs, and we cannot see the prices that customers are presumably reacting to when they switch subscriptions. We cannot see whether the EV customers are subscribing to standard CONSUMPTION tariffs or to ELECTRIC_VEHICLE tariffs.

I can imagine this type of info being displayed in the form of collapsible hierarchical lists, with tables at the bottom level showing tariff terms and rates. Of course, the content of these lists will change whenever subscriptions change (no need to show inactive tariffs -- no subscribers).

A by-broker list might be structured as
.brokers
...PowerTypes
.....active tariffs sorted by number of subscribers -- display shows net change in subscribers after the last tariff-publication event.
.......selecting a tariff opens up a tabular description of the tariff its rates, and perhaps the revenue generated so far by this tariff
.......selecting the subscriber-list shows subscriber types and associated populations

A by-PowerType list would just leave off the broker level and include broker info in the individual tariffs.

These lists would change by timeslot. Perhaps the arrow keys move forward and backward by timeslot, or to the next or previous slot where subscription changes show up.

**Comment by erikkemperman** [comment]
I've made a card for this in our issue tracker, to be considered by the next Dept. 1 priority meeting.
